api/http/api_inventory.go

```diff
@@ -477,6 +471,72 @@ func (i *inventoryHandlers) GetDevicesByGroup(w rest.ResponseWriter, r *rest.Req
 	w.WriteJson(ids)
 }
 
+func (i *inventoryHandlers) AppendDevicesToGroup(w rest.ResponseWriter, r *rest.Request) {
+	var deviceIDs []model.DeviceID
+	ctx := r.Context()
+	l := log.FromContext(ctx)
+	groupName := model.GroupName(r.PathParam("name"))
+	if err := groupName.Validate(); err != nil {
+		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		return
+	}
+
+	if err := r.DecodeJsonPayload(&deviceIDs); err != nil {
+		u.RestErrWithLog(w, r, l,
+			errors.Wrap(err, "invalid payload schema"),
+			http.StatusBadRequest,
+		)
+		return
+	} else if len(deviceIDs) == 0 {
+		u.RestErrWithLog(w, r, l,
+			errors.New("no device IDs present in payload"),
+			http.StatusBadRequest,
+		)
+		return
+	}
+	updated, err := i.inventory.UpdateDevicesGroup(
+		ctx, deviceIDs, groupName,
+	)
+	if err != nil {
+		u.RestErrWithLogInternal(w, r, l, err)
+		return
+	}
+	w.WriteJson(updated)
+}
+
+func (i *inventoryHandlers) ClearDevicesGroup(w rest.ResponseWriter, r *rest.Request) {
+	var deviceIDs []model.DeviceID
+	ctx := r.Context()
+	l := log.FromContext(ctx)
+
+	groupName := model.GroupName(r.PathParam("name"))
+
+	if err := r.DecodeJsonPayload(&deviceIDs); err != nil {
+		u.RestErrWithLog(w, r, l,
+			errors.Wrap(err, "invalid payload schema"),
+			http.StatusBadRequest,
+		)
+		return
+	} else if len(deviceIDs) == 0 {
+		u.RestErrWithLog(w, r, l,
+			errors.New("no device IDs present in payload"),
+			http.StatusBadRequest,
+		)
+		return
+	} else if err = groupName.Validate(); err != nil {
```

Comment:
just nitpick: I would move the validation to line 513, right after the `groupName := ...`, if the group is not valid then there is no point in decoding the payload.